compiler/src/main/java/com/squareup/anvil/compiler/codegen/ContributesSubcomponentHandlerGenerator.kt

```diff
@@ -501,7 +501,14 @@ internal fun ClassId.generatedAnvilSubcomponent(parentClass: ClassId): ClassId {
   val parentClassPackageSuffix = if (parentClass.relativeClassName.isRoot) {
     "root"
   } else {
-    parentClass.relativeClassName.asString().lowercase()
+    // We shorten the parent class name even further. Kee all uppercase letters, digits and
+    // lowercase letters after an uppercase letter to avoid potential conflicts.
+    val className = parentClass.relativeClassName.asString()
+    className
+      .filterIndexed { index, c ->
+        index == 0 || c.isUpperCase() || c.isDigit() || className[index - 1].isUpperCase()
```

Comment:
Were you able to get a sense of how close this gets to the name limit compared to, for example, taking up to two characters after an upper case letter? I could see conflicts being pretty common when only taking up to two characters per word

Reply:
Reverted entirely. It was to risky like you noted.